src/front/wgsl/mod.rs

```diff
@@ -1843,19 +1866,26 @@ impl Parser {
                                 arguments,
                             }
                         }
-                    });
-                }
+                    })
+                } else {
+                    None
+                };
                 lexer.expect(Token::Paren(')'))?;
                 lexer.expect(Token::Paren('{'))?;
 
                 block.extend(initialization);
 
-                let mut body = [crate::Statement::If {
-                    condition,
-                    accept: Vec::new(),
-                    reject: [crate::Statement::Break].to_vec(),
-                }]
-                .to_vec();
+                let mut body = Vec::new();
+                if let Some(condition) = condition {
+                    body.push(crate::Statement::If {
+                        condition: context.expressions.append(crate::Expression::Unary {
```

Comment:
why is this better than the previous variant? It looks like we are adding an extra expression `Unary` without actually reducing any other parts

Reply:
This way it conforms literally to the loop transformation example [here](https://gpuweb.github.io/gpuweb/wgsl.html#for-statement). I wasn't sure about this either (- how normative are the examples?) - I'll happily remove that commit if preferred.

Reply:
These IRs are equivalent anyway, we aren't required to be literal here. If the difference is 1 expression, we should pick IR that has less of them.